### ISSUE TITLE
Issues when using nested selectors in a postcss-mixin

### ISSUE THREAD

**Issue by spacedawwwg** [postcss/postcss-nested#101]
Using:  `postcss@8.1.1`, `postcss-nested@5.0.1` and `postcss-mixins@7.0.1` 

postcss.config.js
```
module.exports = { 
  plugins: [
    require('postcss-mixins'),
    require('postcss-nested'),
  ] 
};
```

input.css
```
@define-mixin test {
  color: green;

  .test-child {
    color: red;
  }
}

.test {
  @mixin test;
}
```

output.css
```
.test {
  color: green;

  .test-child {
    color: red;
  }
}
```

expected.css
```
.test {
  color: green;
}
.test .test-child {
   color: red;
}
```

**Comment by spacedawwwg** [comment]
This seems to work fine with `postcss@7.0.35`, `postcss-nested@4.2.3` and `postcss-mixins@6.2.3`

**Comment by ai** [comment]
Here is an issue where we discussed the problem https://github.com/postcss/postcss-nested/issues/101#issuecomment-722414324

We need to update `postcss-nested` to use visitor API. Sorry, I do not have time right now. Will be glad for PR.

**Comment by spacedawwwg** [comment]
> Here is an issue where we discussed the problem [#101 (comment)](https://github.com/postcss/postcss-nested/issues/101#issuecomment-722414324)
> 
> We need to update `postcss-nested` to use visitor API. Sorry, I do not have time right now. Will be glad for PR.

@ai Happy to take a look. Not written a postcss plugin before - any chance of being pointed in the right direction of this visitor api?

**Comment by ai** [comment]
To solve this issue we need to replace `Once` in `postcss-nested` to `Rule`/`AtRule` according to Step 3 in [Plugin Migration guide](https://evilmartians.com/chronicles/postcss-8-plugin-migration)

**Comment by dan2k3k4** [comment]
Yeah, and to rewrite the functions being used in `postcss-nested`, I started a bit in the other open issue (perhaps we can close that one / merge it with this one).

@spacedawwwg I provided a patch diff to help get started:

https://github.com/postcss/postcss-nested/issues/81#issuecomment-714028404

**Comment by ai** [comment]
Fixed ec804d1

**Comment by ai** [comment]
Fixed in 5.0.2.

---

Sorry for long delay, but it is hard to work on big ecosystem without proper fundings. If you want to fix the source of the problem talk with manager about sponsoring open source dependencies (here is [PostCSS OpenCollective](https://opencollective.com/postcss/)).
